pkg/client/httputils.go

```diff
@@ -38,6 +38,8 @@ type HTTPHelper struct {
 	RootEndpoint string
 	Dumpster     io.Writer
 	AccessToken  string
+	UseHTTPBasic bool
```

Comment:
let's merge these two into `HTTPBasicUsername` and call `SetBasicAuth` if not empty.